src/proto/mod.rs

```diff
@@ -74,3 +74,4 @@ pub mod pi;
 pub mod rng;
 pub mod security;
 pub mod shim;
+pub mod string;
```

Comment:
shouldn't this module be called "collation"? String is a bit misleading.. what do you think @nicholasbishop ?